### ISSUE TITLE
CCD WCS headers need to be confirmed, fixed

### ISSUE THREAD

**Issue by chazshapiro** [CaltechOpticalObservatories/NGPS#193]
Headers that compute how image pixels are mapped to detector pixels and wavelengths seem wrong.  They need to account for overscans, flips, which amps, etc.

**Comment by astronomerdave** [comment]
Which keywords "seem" wrong, because I think this is already accounted for.

**Comment by chazshapiro** [comment]
So as an example here's ngps_241117_0110.fits (R channel):

```
BIASSEC = '[4214:4314,1:1420]' / overscan section                               
CCDSEC  = '[1:4214,1:1220]'    / physical format of CCD                         
DATASEC = '[1:4214,1:1220]'    / section containing the CCD data                
LTV1    =                  100                                                  
LTV2    =                  200                                                  
OS_COLS =                  100 / overscan cols                                  
OS_ROWS =                  200 / overscan rows     
```
<img width="1057" alt="Screenshot 2024-11-25 at 9 25 33 PM" src="https://github.com/user-attachments/assets/41b27664-8598-420f-8bdd-4508264003a1">

The header says the serial overscan is on the right (4214:4314).  I can't tell if that's correct, but regardless, it's inconsistent with the PHYSICAL X value of the crosshairs, which is negative, meaning the crosshairs are off the edge of the detector, which they should not be if that's not the overscan side.  The PHYSICAL Y value of the crosshairs is also negative but that's clearly not off the edge.

We also need to update the wavelength values for the SPECTRUM X coordinates.  I need to take a look at CF's solution for that.  SPECTRUM Y is only needed for the fancy display which we're not using yet.

**Comment by chazshapiro** [comment]
I think PHYSICAL X/Y is adjusted by LTV keywords

**Comment by chazshapiro** [comment]
```
CDELT1A =                 0.57 / Dispersion in Angstrom/pixel                   
CRPIX1A =                  101                                                  
CRVAL1A =                5610. / Reference value in Angstrom  
```

The CRPIX1A also implies the overscan is on the left, i.e. the wavelength counting starts at image pixel 101.

**Comment by chazshapiro** [comment]
Beyond that, let's change:
CDELT1A=.576. (R)
CDELT1A=.71. (I)

And let's wait to get the other things fixed before adjusting CRVAL1A.

**Comment by chazshapiro** [comment]
This (R, log scale) looks like the serial overscan is on the left 
<img width="1057" alt="Screenshot 2024-11-25 at 10 07 58 PM" src="https://github.com/user-attachments/assets/2d61c847-4817-4aa9-a9e4-e48011075fa9">

**Comment by chazshapiro** [comment]
Yes to both.  The first item in the title is "confirmed" so that's on me.

**Comment by astronomerdave** [comment]
No, BIASSEC and DATASEC are correct.

There are two columns that appear in the readout which are not part of the actual image, which has the effect of shifting the image by 2 columns. You will always get the full image area, however. For example, for 4114 image pixels and 100 bias pixels, the FITS file comes out as 4214 pixels wide, but due to being shifted left by 2 pixels, the overscan runs from column 1 to column 98, the image starts at 99 (instead of 101) and ends at 4212 (instead of 4214). The rightmost 2 columns are bogus. No one needs to remember all of these details as long as they use the headers because this is reflected in the BIASSEC and DATASEC keywords. Turn on "use DATASEC" in ds9 and it will correctly display the image.

**Comment by chazshapiro** [comment]
OK re: BIASSEC and DATASEC

SPECTPART is a header in the spectrograph FITS extensions identifying which end of the spectrum it is (RED, BLUE, or WHOLE).  This is needed for split readouts saved to multiple extensions, otherwise everything is WHOLE.

For the CCD_ID, it would be nice to keep both, e.g. "118235_2_3 (sg2)"

The L and B parameters would be in camerad and used to generate WCS headers like CRPIX1 and LTV1, which don't look correct.  Please check the definitions in software doc 9.2.1 and make sure that they match the code.

**Comment by chazshapiro** [comment]
I edited astrocam.cpp (no commit) with an arithmetic change. (search for ###CHAZ )

Image e.g. ngps_250422_0062.fits does not appear to have parallel overscan, but header has OS_ROWS=100.

CCD_ID still missing serial number.
